lib/index.js

```diff
@@ -177,6 +173,27 @@ exports.request = function (method, url, options, callback, _trace) {
         req.write(options.payload);
     }
 
+    // Custom abort method to detect early aborts
+
+    var _abort = req.abort;
+    var aborted = false;
+    req.abort = function () {
```

Comment:
`abort()` is part of the exposed api in the req object that is returned. It should definitely be supported. Without this patch an early `abort()` call can cause the request callback to never be called.